tutorials-v4/python-introduction/Beginning_Python.md

```diff
@@ -0,0 +1,681 @@
+
+```python
+np
+```
+
+Here we are asking Python to get all of the basic functions (this is what * means) from the NumPy module.  We can now do more impressive calculations:
+
+```python
+np.exp(2.34)
+```
+
+```python
+np.sqrt(5)
+```
+
+```python
+np.sinc(0.5)
+```
+
+## Variables
+
+
+If we want to be able to store the numbers and results from our calculations then we must define variables using the "=" sign:
+
+```python
+radius = 5
+area = np.pi * radius**2
+area
+```
+
+We see that our variables name is defined on the left of the ```=``` sign and the value its given is defined on the right.  Here we have also used the ```pi``` variable that has been predefined by NumPy.  Variables can then be used in other expressions.  
+
+If a predefined variable is again used on the left side of ```=``` then its original value is replaced.
+
+```python
+x = 10
+x = (x**2 + 25) / 10
+x
+```
+
+This is different than the mathematical equation $10x=x^{2}+25$  which has the solution $x=5$.  Therefore, it is important to remember that the ```=``` sign in a computer program is **not** equivalent to the mathematical equality. 
+
+
+What happens if you try to use a variable without first defining it? 
+
+Python would give us an error that the variable is not defined.  In addition, there are several words that are reserved by the Python language and cannot be used as variables:
+
+    and, as, assert, break, class, continue, def, del, elif, else, except, 
+    exec, finally, for, from, global, if, import, in, is, lambda, not, or,
+    pass, print, raise, return, try, while, with, yield
+    
+Other than the above reserved words, your variables can be anything that starts with a letter or the underscore character "$\_$" preceded by any combination of alphanumeric characters and "$\_$".  Note that using upper or lower case letters will give you two different variables.
+
+```python
+_freq = 8
+Oscillator_Energy = 10
+_freq * Oscillator_Energy
+```
+
+### Some Rules About Variables
+
+
+Although there are many ways to define variables in Python, it is best to try to define your variables in all the same way.  In this class, all of our variables will use only lower case characters. 
+
+
+```python
+speed_of_light = 2.9979 * 10**8
+spring_constant = np.sqrt(2 / 5)
+```
+
+It is also good practice to use variable names that correspond to the physical quantity that the variable represents.
+
+
+## Strings
+
+
+Often times we want to print some text along with our variables, ask the user for input, or actually use the words and letters themselves as variables (e.g. in DNA analysis).  All of these can be accomplished using **strings**.  We have already seen one string already in this class: 
+
+```python
+"Hello Class"
+```
+
+We can also use double quotes
+
+```python
+"Hello Class"
+```
```

Comment:
```suggestion
We can also use single quotes, e.g. `'Hello Class'`.
```
I guess some auto formatting changed the example above from single quotes to double?